### PR TITLE
HV-1103 Removing remaining switches on Java version

### PR DESCRIPTION
**Pull request by gunnarmorling**

### CODE REVIEW

File: engine/src/main/java/org/hibernate/validator/internal/metadata/provider/AnnotationMetaDataProvider.java [hunk 1]
```diff
@@ -705,7 +726,119 @@ public ConstraintType getPartition(ConstraintDescriptorImpl<?> v) {
 	 * @return a set of type arguments constraints, or an empty set if no constrained type arguments are found
 	 */
 	protected Set<MetaConstraint<?>> findTypeAnnotationConstraintsForExecutableParameter(Member member, int i) {
-		// To be extended by subclasses
-		return Collections.emptySet();
+		Parameter parameter = ( (Executable) member ).getParameters()[i];
```
Comment (gunnarmorling) [inline engine/src/main/java/org/hibernate/validator/internal/metadata/provider/AnnotationMetaDataProvider.java]: That's not new code, it just has been inlined from the separate sub-class we had before.

File: engine/src/main/java/org/hibernate/validator/internal/util/Version.java [hunk 2]
```diff
@@ -17,15 +14,9 @@
  * @author Kevin Pollet &lt;kevin.pollet@serli.com&gt; (C) 2012 SERLI
  */
 public final class Version {
-	private static final Pattern JAVA_VERSION_PATTERN = Pattern.compile( "^(?:1\\.)?(\\d+)$" );
```
Comment (gsmet) [inline engine/src/main/java/org/hibernate/validator/internal/util/Version.java]: Mmmmh, why do we remove all this? I think we should keep it. We probably will have version specific features in the future and I don't see the point in removing this?

Comment (gunnarmorling) [inline engine/src/main/java/org/hibernate/validator/internal/util/Version.java]: I don't like keeping code which is unused. It's still in the repo should there be the need to resurrect it.

Comment (gsmet) [inline engine/src/main/java/org/hibernate/validator/internal/util/Version.java]: I usually agree with this motto but not in this case. Anyway, I'll merge it and keep my "I'll tell you so" for later :).

**Comment by gsmet** [comment]
Rebased and merged! Thanks, nice cleanup!

**Pull request closed without merging**